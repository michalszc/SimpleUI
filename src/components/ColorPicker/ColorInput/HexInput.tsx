import React, { FC } from "react";
import * as CSS from "csstype";
import CustomInput from "./CustomInput";
import styled from "styled-components";

export interface HexInputProps {
    color: CSS.DataType.Color;
    setColor: (value: CSS.DataType.Color) => void;
}

const HexContainer = styled.div`
    display: grid;
    justify-content: center;
    & > .simpleui-colorpicker-inputcontainer {
        grid-template-columns: 22px 70px;
    }
`;

const HexInput: FC<HexInputProps> = ({
    color, setColor
}) => (
    <HexContainer>
        <CustomInput prefix={'#'} value={color.replace('#', '')} onChange={(value: string) => setColor(`#${value}`)} />
    </HexContainer>
);

export default HexInput;
