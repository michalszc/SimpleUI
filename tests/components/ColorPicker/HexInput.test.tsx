import React from "react";
import '@testing-library/jest-dom';
import { render } from "@testing-library/react";
import HexInput from "../../../src/components/ColorPicker/ColorInput/HexInput";

describe("HSLAInput", () => {
    const color = '#FFF';
    const mockFn = jest.fn();
    test("should properly render HexInput component", () => {
        const container = render(<HexInput color={color} setColor={mockFn}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly set color values", () => {
        const { getByRole } = render(<HexInput color={color} setColor={mockFn}/>);
        const input = getByRole('textbox');
        expect(input).toHaveValue('FFF');
    });
});
