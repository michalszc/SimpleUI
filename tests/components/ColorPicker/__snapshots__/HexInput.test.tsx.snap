// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HSLAInput should properly render HexInput component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-hLBbgP hbVVJF"
      >
        <div
          class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
        >
          <div
            class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
          >
            #
          </div>
          <input
            class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
            value="FFF"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-hLBbgP hbVVJF"
    >
      <div
        class="sc-bcXHqe dLNOCe simpleui-colorpicker-inputcontainer"
      >
        <div
          class="sc-gswNZR icpVtD simpleui-colorpicker-input-prefix"
        >
          #
        </div>
        <input
          class="sc-dkrFOg dGsRDm simpleui-colorpicker-input"
          value="FFF"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
